genesyscloud/team/resource_genesyscloud_team_schema.go

```diff
@@ -0,0 +1,92 @@
+package team
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	registrar "terraform-provider-genesyscloud/genesyscloud/resource_register"
+)
+
+/*
+resource_genesycloud_team_schema.go holds four functions within it:
+
+1.  The registration code that registers the Datasource, Resource and Exporter for the package.
+2.  The resource schema definitions for the team resource.
+3.  The datasource schema definitions for the team datasource.
+4.  The resource exporter configuration for the team exporter.
+*/
+const resourceName = "genesyscloud_team"
+
+// SetRegistrar registers all of the resources, datasources and exporters in the package
+func SetRegistrar(regInstance registrar.Registrar) {
+	regInstance.RegisterResource(resourceName, ResourceTeam())
+	regInstance.RegisterDataSource(resourceName, DataSourceTeam())
+	regInstance.RegisterExporter(resourceName, TeamExporter())
+}
+
+// ResourceTeam registers the genesyscloud_team resource with Terraform
+func ResourceTeam() *schema.Resource {
+	return &schema.Resource{
+		Description: `Genesys Cloud team`,
+
+		CreateContext: gcloud.CreateWithPooledClient(createTeam),
+		ReadContext:   gcloud.ReadWithPooledClient(readTeam),
+		UpdateContext: gcloud.UpdateWithPooledClient(updateTeam),
+		DeleteContext: gcloud.DeleteWithPooledClient(deleteTeam),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The team name",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"division_id": {
+				Description: "The division to which this entity belongs.",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"description": {
+				Description: "Team information.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"member_count": {
+				Description: "Number of members in a team",
+				Optional:    true,
+				Type:        schema.TypeInt,
+			},
+		},
+	}
+}
+
+// TeamExporter returns the resourceExporter object used to hold the genesyscloud_team exporter's config
+func TeamExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: gcloud.GetAllWithPooledClient(getAllAuthTeams),
+		RefAttrs:         map[string]*resourceExporter.RefAttrSettings{
+			// TODO: Add any reference attributes here
```

Comment:
You need to make sure that you reference division_id as an exportable field here.  Otherwise if someone tries to export this object they will get the guid for the division exported instead of a logical name.